homeassistant/components/demo/climate.py

```diff
@@ -259,8 +259,8 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
         ):
             self._target_temperature_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
             self._target_temperature_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
-        if kwargs.get(ATTR_HVAC_MODE) is not None:
-            self._hvac_mode = HVACMode(str(kwargs.get(ATTR_HVAC_MODE)))
+        if (hvac_mode := kwargs.get(ATTR_HVAC_MODE)) is not None:
+            self._hvac_mode = hvac_mode
```

Comment:
You've removed the enumification of the value too, was that intentional?

Reply:
No linters complained, so I guessed it was shorter, easier to read and solved the issue. Should I put them back?

Reply:
The `hvac_mode` parameter is coerced to a `HVACMode` enum by the service schema before passing to this method. So it's ok to not copy it again to an enum.

https://github.com/home-assistant/core/blob/85e782055bc4dc7f114bc6f7f545759d641762d5/homeassistant/components/climate/__init__.py#L127